src/v3/components/time_picker_slider/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { Box, Typography } from '@mui/material';
+import { CPETimePickerSliderProps } from './time_picker_slider.types';
+import {
+  ActiveCaseFilter,
+  CASE_SIZE,
+  CaseContainerStyle,
+  TimePickerArrowStyle,
+  TimePickerCaseStyle,
+  TimePickerContainerStyle,
+  TimePickerSelectorStyle,
+  TimePickerTypography,
+} from './time_picker.styles';
+import { MutableRefObject, useEffect, useRef } from 'react';
+import { KeyboardArrowDown, KeyboardArrowUp } from '@mui/icons-material';
+
+// Used to center the cards
+const BlankCase = () => {
+  return <Box sx={TimePickerCaseStyle} />;
+};
+
+const TimePickerCase = ({ value }: { value: number }) => {
+  return (
+    <Box sx={TimePickerCaseStyle}>
+      <Typography variant="body1" component="span" sx={TimePickerTypography}>
+        {value.toString().padStart(2, '0')}
+      </Typography>
+    </Box>
+  );
+};
+
+const TimePickerHourCases = ({ ampm }: { ampm: boolean }) => {
+  const hours = [];
+  for (let h = 0; h < (ampm ? 12 : 24); h++) {
+    hours.push(<TimePickerCase value={h} />);
+  }
+  return (
+    <>
+      <BlankCase />
+      {hours}
+      <BlankCase />
+    </>
+  );
+};
+
+const TimePickerMinutesCases = () => {
+  const minutes = [];
+  for (let m = 0; m < 60; m++) {
+    minutes.push(<TimePickerCase key={m} value={m} />);
+  }
+
+  return (
+    <>
+      <BlankCase />
+      {minutes}
+      <BlankCase />
+    </>
+  );
+};
+
+const TimePickerSelector = ({ children, value }: { children: React.ReactNode; value: MutableRefObject<string> }) => {
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const disactiveBox = (box: Element) => {
+    if (box.firstElementChild) (box.firstElementChild as HTMLElement).style.color = 'var(--grey-200)';
+  };
+  const activeBox = (box: Element) => {
+    if (box.firstElementChild) (box.firstElementChild as HTMLElement).style.color = 'var(--accent-dark)';
+  };
+
+  // Add active class to the center box
+  const updateActiveBox = () => {
+    const container = containerRef.current;
+    const scrollPosition = container.scrollTop;
+
+    const boxes = container.children;
+    const centerBox = boxes[Math.round(scrollPosition / CASE_SIZE) + 1];
+
+    for (let index = 0; index < boxes.length; index++) {
+      disactiveBox(boxes[index]);
+    }
+    activeBox(centerBox);
+    value.current = centerBox.textContent;
+  };
+
+  // Scroll [unit] case up or down
+  const scroll = (unit: number) => {
+    const container = containerRef.current;
+    const scrollPosition = container.scrollTop;
+    container.scrollTo({ top: scrollPosition + unit * CASE_SIZE, behavior: 'smooth' });
+  };
+
+  useEffect(updateActiveBox);
+
+  return (
+    <div style={TimePickerSelectorStyle}>
+      <div style={ActiveCaseFilter}></div>
+      <div onClick={() => scroll(-1)} style={TimePickerArrowStyle}>
+        <KeyboardArrowUp />
```

Comment:
@Al11o: please confirm also with the designers for the icons that should be used, instead of the default one provided by MUI. But to have the icon button, please use the MUI `IconButton` component.

Reply:
@rhahao the icons are correct, as per Figma design 👍